### ISSUE TITLE
Does UPDATE_FREQ (gradient accumulation) affect the TOTAL_NUM_UPDATES?

### ISSUE THREAD

**Issue by ASoleimaniB** [pytorch/fairseq#3529]
If we set TOTAL_NUM_UPDATES=10 we would have 10 iterations and 10 weight update. What if we have TOTAL_NUM_UPDATES=10, and UPDATE_FREQ=2 (gradient accumulation=2)! Do we have still 10 updates (or 5)? (assume MAX_TOKENS is fixed)
